specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,389 @@
+| GUID_POWERSCHEME_PERSONALITY      | The active power scheme personality has changed. All power schemes map to one of these personalities.                                                                                                                  | `PowerSchemePersonality`                                               |
+| GUID_SESSION_DISPLAY_STATUS       | The display associated with the application's session has been powered on or off.                                                                                                                                      | Covered by GUID_CONSOLE_DISPLAY_STATE.                                 |
+| GUID_SESSION_USER_PRESENCE        | The user status associated with the application's session has changed.                                                                                                                                                 | `UserPresenceStatus`                                                   |
+| GUID_SYSTEM_AWAYMODE              | The system is entering or exiting away mode.                                                                                                                                                                           | `SystemAwayModeStatus` |
+
+The existing PowerManager class exposes only static properties and events: all members are valid to
+be brought over to the Reunion version. Many existing enumerated values are brought over as-is as
+well.
+
+```idl
+namespace Microsoft.Windows.System.Power
+{
+    // Clone to Windows.System.Power.BatteryStatus
+    enum BatteryStatus {
+        NotPresent,
+        Discharging,
+        Idle,
+        Charging
+    }
+
+    // Clone of Windows.System.Power.EnergySaverStatus
+    enum EnergySaverStatus {
+        Disabled,
+        Off,
+        On
+    }
+
+    // Clone of Windows.System.Power.PowerSupplyStatus
+    enum PowerSupplyStatus {
+        NotPresent,
+        Inadequate,
+        Adequate
+    }
+
+    // Status of the device's power source
+    enum PowerSourceKind {
+        AC,
+        DC,
+        ShortTerm
+    }
+
+    // State of the primary display
+    enum DisplayState {
+        Off,
+        On,
+        Dimmed
+    }
+
+    // Indicates whether the system is busy or idle.
+    enum SystemBusyState {
+        Idle,
+        Busy,
+    }
+
+    // Active power management personalities supported by the system
+    enum PowerSchemePersonality {
+        HighPerformance,
+        PowerSaver,
+        Balanced
+    }
+
+    // User presence indicator - whether the user is present or absent
+    // (active or idle)
+    enum UserPresenceStatus {
+        Present,
+        Absent
+    }
+
+    // Indicates whether the system is entering or exiting "away" mode
+    enum SystemAwayModeStatus {
```

Comment:
Entering/Exiting is a transition, so not meaningful for a status query. (Unless the system is always "entering" and "exiting" away mode and never quite gets there or leaves?) Should be something like

```csharp
enum SystemAwayMode
{
    NotAway,
    Away,
}
```

Reply:
DISCUSSION: Many of these names were straight conversions from the GUID forms of the power state identifiers. 

RECOMMEND: Use the `SystemAwayMode` listed.

Reply:
Fixed.